### PR TITLE
Feature/connected

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/org/sonarlint/cli/SonarLint.java [hunk 1]
```diff
@@ -37,20 +37,25 @@
...
+  private final StandaloneSonarLintEngine engine;
 
-  public SonarLint(SonarLintClient client) throws IOException {
-    this.client = client;
+  public SonarLint(StandaloneSonarLintEngine engine) throws IOException {
+    this.engine = engine;
+    this.running = true;
   }
 
   public static SonarLint create(Options opts) throws IOException {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/cli/SonarLint.java]: MAJOR Remove this unused method parameter "opts". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1172)

File: src/main/java/org/sonarlint/cli/report/Reporter.java [hunk 2]
```diff
@@ -22,9 +22,9 @@
 import java.util.Date;
 import java.util.List;
 
-import org.sonarsource.sonarlint.core.AnalysisResults;
-import org.sonarsource.sonarlint.core.IssueListener;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.AnalysisResults;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.Issue;
 
 public interface Reporter {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/cli/report/Reporter.java]: MINOR Annotate the "Reporter" interface with the @FunctionalInterface annotation [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1609)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/main/java/org/sonarlint/cli/SonarLint.java [hunk 3]
```diff
@@ -61,13 +66,15 @@ public static SonarLint create(Options opts) throws IOException {
     } catch (Exception e) {
       throw new IllegalStateException("Error loading plugins", e);
     }
-    SonarLintClient c = SonarLintClient.builder()
+
+    StandaloneGlobalConfiguration config = StandaloneGlobalConfiguration.builder()
       .addPlugins(plugins)
       .setLogOutput(new DefaultLogOutput(LOGGER))
-      .setVerbose(opts.isVerbose())
+      // verbose?
```
Comment (henryju) [inline src/main/java/org/sonarlint/cli/SonarLint.java]: IIRC we dropped this parameter to always have DEBUG logs in the callback and filter on client side

**Comment by henryju** [comment]
LGTM

**Pull request merged**